### PR TITLE
More robust function parsing with pglast, procedure support

### PR DESCRIPTION
**Pull request by dani-maarouf**
- Use pglast to parse functions into signature and definition
- Add support for procedures

Caveats:
- pglast parsing was only added for functions/procedures, not any other kind of entity
- functions formatting and comments are not preserved
- SQL constructs with equivalent representations may be transformed. e.g. `STRICT` may be changed to `RETURNS NULL ON NULL INPUT`

### CODE REVIEW

**Comment by olirice** [comment]
thanks for opening

Unfortunately, I don't want to add support for procedures directly in alembic_utils. Functions have proven to be problematic in that they are hard to parse reliably, support multiple PLs etc etc. For that reason I'm hesitant to expand into procedures, which will face those same problems.

alembic_utils is extensible though, so you can certainly maintain this in your own fork, or subclass `ReplaceableEntity` directly in the project where you want to use it

**Comment by dani-maarouf** [comment]
Sure. Thanks for writing this utility, it's quite useful.

In my fork I've added a few features that I need, detailed here: https://github.com/dani-maarouf/alembic_utils?tab=readme-ov-file#alembic-utils

Let me know if you want me to open a PR to upstream any of the changes
